sbt-scalajs-bundler/src/main/scala/scalajsbundler/util/JS.scala

```diff
@@ -79,17 +93,35 @@ object JS {
 
   def `new`(ctor: JS, args: JS*): JS = JS(New(ctor.tree, args.map(_.tree).to[List]))
 
-  def toJson(obj: JS): String = show(obj.tree, isStat = false)
-
-  def show(tree: Tree, isStat: Boolean = true): String = {
-    val writer = new java.io.StringWriter
-    val printer = new Printers.JSTreePrinter(writer)
-    printer.printTree(tree, isStat)
-    writer.toString
-  }
-
   private val identifierSeq = new AtomicInteger(0)
   private def freshIdentifier(): String =
     s"x${identifierSeq.getAndIncrement()}"
 
 }
+
+final class JSON(tree: Tree) extends JSLike(tree)
+
+object JSON {
+
+  implicit lazy val position: Position = Position.NoPosition
+
+  def apply(tree: Tree): JSON = new JSON(tree)
+
+  /** Array literal. */
+  def arr(elems: JSON*): JSON = JSON(ArrayConstr(elems.map(_.tree).to[List]))
+
+  /** Boolean literal */
+  def bool(value: Boolean): JSON = JSON(BooleanLiteral(value))
+
+  /** Object literal */
+  def obj(fields: (String, JSON)*): JSON =
```

Comment:
That crossed my mind at first but that ends up allowing mixing of `JS` and `JSON` which we wanted to avoid, right?. For example if `JSON.obj` accepts `JSLike` key/values as you suggest, this compiles:

```scala
scala> val x = JSON.obj("a" -> JSON.bool(true), "b" -> JS.regex("[abAZ]"))
x: scalajsbundler.util.JSON = scalajsbundler.util.JSON@1084acc1
```